src/icetransport.cpp

```diff
@@ -145,7 +145,10 @@ Description IceTransport::getLocalDescription(Description::Type type) const {
 }
 
 void IceTransport::setRemoteDescription(const Description &description) {
-	mRole = description.role() == Description::Role::Active ? Description::Role::Passive
+    if (description.role() == Description::Role::ActPass)
```

Comment:
This almost certainly is not right. I had to add this test because when the browser creates the first offer, and then I renegotiate with a new offer, the browser will reject the offer because the offer's role was active.

Reply:
I see the issue, however this will make the answerer default to passive, which is not recommended as it is inefficient. 
I think it would be better to change the return line of `IceTransport::getLocalDescription` like this:
```
return Description(string(sdp), type, type == Description::Type::Offer ? Description::Role::ActPass : mRole);
```
The modification should also be done for the libnice implementation.

Reply:
Just to confirm, I am undoing what I have done, and then changing the result of `IceTransport::getLocalDescription` to what you have said.

Reply:
Yes, it should work fine.